File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        res = count = i = 0
        for num in nums:
            if num & 1:
                k -= 1
                count = 0
            
            while(not k):
                k += nums[i] & 1
                count += 1
                i += 1
            
            res += count
        
        return res
        
        
            
            
            
            

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.go
func numberOfSubarrays(nums []int, k int) int {
    var oddCounts = make(map[int]int)
    oddCounts[0] = 1
    var currCounts, res = 0, 0
    for _, num := range nums {
        if num % 2 == 1 {
            currCounts++
        }
        oddCounts[currCounts]++
        if oddCounts[currCounts - k] > 0 {
            res += oddCounts[currCounts - k]
        }
    }
    
    return res
}